src/cargo/ops/registry.rs

```diff
@@ -183,6 +187,10 @@ pub fn publish(ws: &Workspace<'_>, opts: &PublishOpts<'_>) -> CargoResult<()> {
         reg_id,
         opts.dry_run,
     )?;
+    if !opts.dry_run {
+        let timeout = std::time::Duration::from_secs(300);
```

Comment:
Wanted to call out the current timeout.

Reply:
I feel like 5 minutes is probably a little too long.  I would think 1 minute is usually more than long enough (assuming https://github.com/rust-lang/crates.io/pull/5200 is merged), and if it takes longer than 1 minute then it is going to take *way* longer (like there is a GitHub outage, or a major internal failure on the heroku side). Unless you have observed significantly longer delays under normal operation?  I wonder if crates.io keeps track of how long that queue usually takes.